ext/TensorKitChainRulesCoreExt.jl

```diff
@@ -539,7 +539,7 @@ function qr_pullback!(ΔA::AbstractMatrix, Q::AbstractMatrix, R::AbstractMatrix,
     ΔR1 = view(ΔR, 1:p, :)
     ΔR11 = view(ΔR, 1:p, 1:p)
 
-    M = similar(R, (p, p))
+    M = zeros(eltype(R), (p, p))
     ΔR isa AbstractZero || mul!(M, ΔR1, R1')
     ΔQ isa AbstractZero || mul!(M, Q1', ΔQ1, -1, +1)
```

Comment:
Great catch!
I think something like this might even be able to avoid the extra zeroing out:
```suggestion
    M = similar(R, (p, p))
    ΔR isa AbstractZero || mul!(M, ΔR1, R1')
    ΔQ isa AbstractZero || mul!(M, Q1', ΔQ1, -1, !(ΔR isa AbstractZero))
```